wagtail/blocks/stream_block.py

```diff
@@ -75,9 +75,15 @@ def as_json_data(self):
 
 
 class BaseStreamBlock(Block):
+    # This is set to `StreamValue` a little further down. The idea is that
+    # developers can override this on custom subclasses to change the default
+    # `value_class` for all instances
+    default_value_class = None
+
     def __init__(self, local_blocks=None, search_index=True, **kwargs):
         self._constructor_kwargs = kwargs
         self.search_index = search_index
+        self.value_class = kwargs.get("value_class", self.default_value_class)
```

Comment:
We should define this as part of the inner `Meta` class, [like BaseStructBlock does](https://github.com/wagtail/wagtail/blob/1ec62a6c2d99365886e41c517d386d8d36fa5d50/wagtail/blocks/struct_block.py#L367) - that way, subclasses can define this in their own Meta class, which is probably a more natural way of working than passing it to the StructBlock constructor.